### PR TITLE
Add more logs to the SDKs

### PR DESCRIPTION
**Pull request by KacperKluka**
I've added a bunch of logs to the SDKs:
- Publisher lifecycle logs
- Subscriber lifecycle logs
- Error logging for the Ably wrapper

### CODE REVIEW

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/PublisherBuilder.kt [hunk 1]
```diff
@@ -71,7 +74,9 @@ internal data class PublisherBuilder(
 
     @RequiresPermission(anyOf = [ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION])
     override fun start(): Publisher {
+        logHandler?.v("$TAG Creating a publisher instance")
```
Comment (QuintinWillison) [inline publishing-sdk/src/main/java/com/ably/tracking/publisher/PublisherBuilder.kt:77]: Looking at this message in context of required fields being missing then I think that might cause confusion when debugging. Perhaps consider moving this line to after the required fields check.

File: subscribing-sdk/src/main/java/com/ably/tracking/subscriber/SubscriberBuilder.kt [hunk 2]
```diff
@@ -26,14 +29,17 @@ internal data class SubscriberBuilder(
         this.copy(logHandler = logHandler)
 
     override suspend fun start(): Subscriber {
+        logHandler?.v("$TAG Creating a subscriber instance")
```
Comment (QuintinWillison) [inline subscribing-sdk/src/main/java/com/ably/tracking/subscriber/SubscriberBuilder.kt:32]: Looking at this message in context of required fields being missing then I think that might cause confusion when debugging. Perhaps consider moving this line to after the required fields check.

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/PublisherBuilder.kt [hunk 3]
```diff
@@ -71,7 +74,9 @@ internal data class PublisherBuilder(
...
         if (isMissingRequiredFields()) {
+            logHandler?.v("$TAG Creating a publisher instance failed due to missing required fields")
```
Comment (QuintinWillison) [inline publishing-sdk/src/main/java/com/ably/tracking/publisher/PublisherBuilder.kt:79]: I generally avoid log-before-throw as more often than not it ends up with a situation where code above this also logs the exception and then when debugging it's difficult to ascertain how many exceptions were actually thrown - whether caught or uncaught.

Therefore, I think you should remove this log statement entirely. It an app developer has managed to call `start()` but is unaware that it's thrown them back an exception then I think they have larger problems afoot. 😬

File: subscribing-sdk/src/main/java/com/ably/tracking/subscriber/SubscriberBuilder.kt [hunk 4]
```diff
@@ -26,14 +29,17 @@ internal data class SubscriberBuilder(
...
         if (isMissingRequiredFields()) {
+            logHandler?.v("$TAG Creating a subscriber instance failed due to missing required fields")
```
Comment (QuintinWillison) [inline subscribing-sdk/src/main/java/com/ably/tracking/subscriber/SubscriberBuilder.kt:34]: Similar to my comment on publisher `start()`, I would suggest removing this log line completely. The app should be catching the exception.

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
Furthermore, on the subject of logging failure/exception cases, I'm a bit worried that the `onError` implementations (`wrapInResultCallback`) that you've provided will also end up with confusing log output when the application itself also logs exceptions. There's also no way for the app developer to opt out of this behaviour, so my gut feeling is that this presents over-logging.

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/PublisherBuilder.kt (hunk 3 above)
Comment (QuintinWillison) [reply publishing-sdk/src/main/java/com/ably/tracking/publisher/PublisherBuilder.kt:79]: I'm removing this objection, given I've now [created an issue](https://github.com/ably/ably-asset-tracking-android/pull/813#ref-issue-1454514942) to look at this across all of our SDKs.

File: subscribing-sdk/src/main/java/com/ably/tracking/subscriber/SubscriberBuilder.kt (hunk 4 above)
Comment (QuintinWillison) [reply subscribing-sdk/src/main/java/com/ably/tracking/subscriber/SubscriberBuilder.kt:34]: I'm removing this objection, given I've now [created an issue](https://github.com/ably/ably-asset-tracking-android/pull/813#ref-issue-1454514942) to look at this across all of our SDKs.

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/PublisherBuilder.kt (hunk 1 above)
Comment (KacperKluka) [reply publishing-sdk/src/main/java/com/ably/tracking/publisher/PublisherBuilder.kt]: [0c143bd](https://github.com/ably/ably-asset-tracking-android/pull/813/commits/0c143bd404aa48d678d6b27fd29b8a82d8b31675)

File: subscribing-sdk/src/main/java/com/ably/tracking/subscriber/SubscriberBuilder.kt (hunk 2 above)
Comment (KacperKluka) [reply subscribing-sdk/src/main/java/com/ably/tracking/subscriber/SubscriberBuilder.kt]: [0c143bd](https://github.com/ably/ably-asset-tracking-android/pull/813/commits/0c143bd404aa48d678d6b27fd29b8a82d8b31675)

**Review by QuintinWillison — APPROVED** [review]

**Pull request merged**